source/common/quic/envoy_quic_utils.cc

```diff
@@ -167,7 +167,7 @@ Network::ConnectionSocketPtr
 createConnectionSocket(const Network::Address::InstanceConstSharedPtr& peer_addr,
                        Network::Address::InstanceConstSharedPtr& local_addr,
                        const Network::ConnectionSocket::OptionsSharedPtr& options,
-                       const bool prefer_gro) {
+                       const bool prefer_gro, const bool connect) {
   if (local_addr == nullptr) {
     local_addr = Network::Utility::getLocalAddress(peer_addr->ip()->version());
```

Comment:
It's better not to call getLocalAddress() here and in the caller of this method. And gating bind() below on local_addr != nullptr.

Reply:
Done, I also changes this so we don't call getLocalAddress if local_addr is null and instead just gets the loopback address so we don't have to call the expensive `getifaddrs`, since all we need this address for is to construct the IoHandle.